From 30f02e2063fe39cd3da9ce4805c1e71f8ecba347 Mon Sep 17 00:00:00 2001
From: Lionel Landwerlin <lionel.g.landwerlin@intel.com>
Date: Mon, 5 Jun 2017 11:06:28 +0100
Subject: [PATCH 2/3] intel: common: add flag to identify platforms by name

The perf infrastructure needs to identify specific platforms, not just
generations.

Upstream-Status: Backport

Signed-off-by: Lionel Landwerlin <lionel.g.landwerlin@intel.com>
Reviewed-by: Kenneth Graunke <kenneth@whitecape.org>
Signed-off-by: Liwei Song <liwei.song@windriver.com>
---
 src/intel/common/gen_device_info.c | 26 ++++++++++++++++++++------
 src/intel/common/gen_device_info.h |  4 ++++
 2 files changed, 24 insertions(+), 6 deletions(-)

diff --git a/src/intel/common/gen_device_info.c b/src/intel/common/gen_device_info.c
index c59580286ab5..c0a64a0ab8a2 100644
--- a/src/intel/common/gen_device_info.c
+++ b/src/intel/common/gen_device_info.c
@@ -304,6 +304,7 @@ static const struct gen_device_info gen_device_info_hsw_gt3 = {
 
 static const struct gen_device_info gen_device_info_bdw_gt1 = {
    GEN8_FEATURES, .gt = 1,
+   .is_broadwell = true,
    .num_slices = 1,
    .l3_banks = 2,
    .max_cs_threads = 42,
@@ -324,6 +325,7 @@ static const struct gen_device_info gen_device_info_bdw_gt1 = {
 
 static const struct gen_device_info gen_device_info_bdw_gt2 = {
    GEN8_FEATURES, .gt = 2,
+   .is_broadwell = true,
    .num_slices = 1,
    .l3_banks = 4,
    .max_cs_threads = 56,
@@ -344,6 +346,7 @@ static const struct gen_device_info gen_device_info_bdw_gt2 = {
 
 static const struct gen_device_info gen_device_info_bdw_gt3 = {
    GEN8_FEATURES, .gt = 3,
+   .is_broadwell = true,
    .num_slices = 2,
    .l3_banks = 8,
    .max_cs_threads = 56,
@@ -419,7 +422,6 @@ static const struct gen_device_info gen_device_info_chv = {
 
 #define GEN9_LP_FEATURES                           \
    GEN9_FEATURES,                                  \
-   .is_broxton = 1,                                \
    .gt = 1,                                        \
    .has_llc = false,                               \
    .num_slices = 1,                                \
@@ -466,6 +468,7 @@ static const struct gen_device_info gen_device_info_chv = {
 
 static const struct gen_device_info gen_device_info_skl_gt1 = {
    GEN9_FEATURES, .gt = 1,
+   .is_skylake = true,
    .num_slices = 1,
    .l3_banks = 2,
    .urb.size = 192,
@@ -473,18 +476,21 @@ static const struct gen_device_info gen_device_info_skl_gt1 = {
 
 static const struct gen_device_info gen_device_info_skl_gt2 = {
    GEN9_FEATURES, .gt = 2,
+   .is_skylake = true,
    .num_slices = 1,
    .l3_banks = 4,
 };
 
 static const struct gen_device_info gen_device_info_skl_gt3 = {
    GEN9_FEATURES, .gt = 3,
+   .is_skylake = true,
    .num_slices = 2,
    .l3_banks = 8,
 };
 
 static const struct gen_device_info gen_device_info_skl_gt4 = {
    GEN9_FEATURES, .gt = 4,
+   .is_skylake = true,
    .num_slices = 3,
    .l3_banks = 12,
    /* From the "L3 Allocation and Programming" documentation:
@@ -500,11 +506,13 @@ static const struct gen_device_info gen_device_info_skl_gt4 = {
 
 static const struct gen_device_info gen_device_info_bxt = {
    GEN9_LP_FEATURES,
+   .is_broxton = true,
    .l3_banks = 2,
 };
 
 static const struct gen_device_info gen_device_info_bxt_2x6 = {
    GEN9_LP_FEATURES_2X6,
+   .is_broxton = true,
    .l3_banks = 1,
 };
 /*
@@ -573,12 +581,14 @@ static const struct gen_device_info gen_device_info_kbl_gt4 = {
 
 static const struct gen_device_info gen_device_info_glk = {
    GEN9_LP_FEATURES,
+   .is_geminilake = true,
    .l3_banks = 2,
 };
 
 /*TODO: Initialize l3_banks when we know the number. */
 static const struct gen_device_info gen_device_info_glk_2x6 = {
-   GEN9_LP_FEATURES_2X6
+   GEN9_LP_FEATURES_2X6,
+   .is_geminilake = true,
 };
 
 #define GEN10_HW_INFO                               \
@@ -609,22 +619,26 @@ static const struct gen_device_info gen_device_info_glk_2x6 = {
 
 static const struct gen_device_info gen_device_info_cnl_2x8 = {
    /* GT0.5 */
-   GEN10_FEATURES(1, 1, 2)
+   GEN10_FEATURES(1, 1, 2),
+   .is_cannonlake = true,
 };
 
 static const struct gen_device_info gen_device_info_cnl_3x8 = {
    /* GT1 */
-   GEN10_FEATURES(1, 1, 3)
+   GEN10_FEATURES(1, 1, 3),
+   .is_cannonlake = true,
 };
 
 static const struct gen_device_info gen_device_info_cnl_4x8 = {
    /* GT 1.5 */
-   GEN10_FEATURES(1, 2, 6)
+   GEN10_FEATURES(1, 2, 6),
+   .is_cannonlake = true,
 };
 
 static const struct gen_device_info gen_device_info_cnl_5x8 = {
    /* GT2 */
-   GEN10_FEATURES(2, 2, 6)
+   GEN10_FEATURES(2, 2, 6),
+   .is_cannonlake = true,
 };
 
 bool
diff --git a/src/intel/common/gen_device_info.h b/src/intel/common/gen_device_info.h
index 620763051948..4a467cca3eff 100644
--- a/src/intel/common/gen_device_info.h
+++ b/src/intel/common/gen_device_info.h
@@ -39,9 +39,13 @@ struct gen_device_info
    bool is_ivybridge;
    bool is_baytrail;
    bool is_haswell;
+   bool is_broadwell;
    bool is_cherryview;
+   bool is_skylake;
    bool is_broxton;
    bool is_kabylake;
+   bool is_geminilake;
+   bool is_cannonlake;
 
    bool has_hiz_and_separate_stencil;
    bool must_use_separate_stencil;
-- 
2.7.4

